src/test/java/org/opensearch/knn/index/query/KNNQueryBuilderTests.java

```diff
@@ -127,6 +127,51 @@ public void testFromXcontent_WithFilter() throws Exception {
         actualBuilder.equals(knnQueryBuilder);
     }
 
+    public void testFromXContent_invalidQueryVectorType() throws Exception {
+        final ClusterService clusterService = mockClusterService(Version.CURRENT);
+
+        final KNNClusterUtil knnClusterUtil = KNNClusterUtil.instance();
+        knnClusterUtil.initialize(clusterService);
+
+        String[] invalidTypeQueryVector = { "a", "b", "c", "d" };
```

Comment:
I would probably include multiple types like string, null and numbers

Reply:
Updated